### PR TITLE
fixbug(zlx): random_collect compatible to episode collector

### PR DESCRIPTION
**Pull request by LuciusMos**
## Description
Fix bug of inital random collect, make it compatible to episode collector.

Use a `random_collect_fn` to replace all corresponding codes in entries.

## Related Issue
https://github.com/opendilab/DI-engine/issues/184

## TODO
@puyuan1996 

td3_vae(different from other entries)

reward_model_ngu(TODO on `collector_kwargs`)

## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/entry/__init__.py [hunk 1]
```diff
@@ -20,3 +20,4 @@
 from .application_entry_trex_collect_data import trex_collecting_data, collect_episodic_demo_data_for_trex
 from .serial_entry_guided_cost import serial_pipeline_guided_cost
 from .serial_entry_gail import serial_pipeline_gail
+from .random_collect import random_collect_fn
```
Comment (PaParaZz1) [inline ding/entry/__init__.py:23]: utils.py can be better

File: ding/entry/random_collect.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+from ding.policy import PolicyFactory
+
+def random_collect_fn(policy_cfg, policy, collector, collector_env, commander, replay_buffer, mark_not_expert=False):
```
Comment (PaParaZz1) [inline ding/entry/random_collect.py:3]: add python typing lint

File: ding/entry/random_collect.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    assert policy_cfg.random_collect_size > 0
```
Comment (PaParaZz1) [inline ding/entry/random_collect.py:4]: add unittest for this function, both episode and sample collector, both transition_with_policy_data True or False

**Review by PaParaZz1 — APPROVED** [review]

File: ding/entry/__init__.py (hunk 1 above)
Comment (LuciusMos) [reply ding/entry/__init__.py]: Resolved. Also rename `random_collect_fn` to `random_collect`

File: ding/entry/random_collect.py (hunk 3 above)
Comment (LuciusMos) [reply ding/entry/random_collect.py]: Resolved. Also add whether to use postprocess data function

**Pull request merged**